API.md

```diff
@@ -0,0 +1,335 @@
+
+```typescript
+/* eslint-disable no-undef */
+interface WizardRequestBase {
+  subWizard?: boolean;
+}
+
+export interface EditWizardRequest extends WizardRequestBase {
+  element: Element;
+}
+
+export interface CreateWizardRequest extends WizardRequestBase {
+  parent: Element;
+  tagName: string;
+}
+
+export type WizardRequest = EditWizardRequest | CreateWizardRequest;
+
+type EditWizardEvent = CustomEvent<EditWizardRequest>;
+type CreateWizardEvent = CustomEvent<CreateWizardRequest>;
+export type WizardEvent = EditWizardEvent | CreateWizardEvent;
+
+type CloseWizardEvent = CustomEvent<WizardRequest>;
+
+export function newEditWizardEvent(
+  element: Element,
+  subWizard?: boolean,
+  eventInitDict?: CustomEventInit<Partial<EditWizardRequest>>
+): EditWizardEvent {
+  return new CustomEvent<EditWizardRequest>('oscd-edit-wizard-request', {
+    bubbles: true,
+    composed: true,
+    ...eventInitDict,
+    detail: { element, subWizard, ...eventInitDict?.detail },
+  });
+}
+
+export function newCreateWizardEvent(
+  parent: Element,
+  tagName: string,
+  subWizard?: boolean,
+  eventInitDict?: CustomEventInit<Partial<CreateWizardRequest>>
+): CreateWizardEvent {
+  return new CustomEvent<CreateWizardRequest>('oscd-create-wizard-request', {
+    bubbles: true,
+    composed: true,
+    ...eventInitDict,
+    detail: {
+      parent,
+      tagName,
+      subWizard,
+      ...eventInitDict?.detail,
+    },
+  });
+}
+
+export function newCloseWizardEvent(
+  wizard: WizardRequest,
+  eventInitDict?: CustomEventInit<Partial<WizardRequest>>
+): CloseWizardEvent {
+  return new CustomEvent<WizardRequest>('oscd-close-wizard', {
+    bubbles: true,
+    composed: true,
+    ...eventInitDict,
+    detail: wizard,
+  });
+}
+
+declare global {
+  interface ElementEventMap {
+    ['oscd-edit-wizard-request']: EditWizardRequest;
+    ['oscd-create-wizard-request']: CreateWizardRequest;
+    ['oscd-close-wizard']: WizardEvent;
+  }
+}
+```
+
+### `ConfigurePluginEvent`
+
+The **configure plugin event** allows the plugin to request that OpenSCD core add, remove, or reconfigure a plugin.
+
+```typescript
+export type ConfigurePluginDetail = {
+  name: string;
+  kind: 'menu' | 'editor';
+  config: Plugin | null;
+};
+
+export type ConfigurePluginEvent = CustomEvent<ConfigurePluginDetail>;
+
+export function newConfigurePluginEvent(name: string, kind: 'menu' | 'editor', config: Plugin | null): ConfigurePluginEvent {
+  return new CustomEvent<ConfigurePluginDetail>('oscd-configure-plugin', {
+    bubbles: true,
+    composed: true,
+    detail: { name, kind, config },
+  });
+}
+```
+
+The combination of `name` and `kind` uniquely identifies the plugin to be configured. If `config` is `null`, the plugin is removed. Otherwise, the plugin is added or reconfigured.
+
+## Theming
```

Comment:
I have done some experimenting with Material 3 (in particular the latest version of [material-web](https://github.com/material-components/material-web) which takes a different approach to theming than what was done in Material 2.

One of the nice features was the way it dealt with light and dark modes (me being a fan of dark mode).
I am curious to know if there was any consideration for migrating to Material 3 (where possible using offical components, noting the team have not managed to complete all of them before going into maintenance mode)?
It was probably be a bit of refactoring amongst all of the plugins that are available.
Many of them at present are not really suited for dark mode.

Reply:
Sounds like a great idea, but I wouldn't include a technical detail about the implementation in these API docs themselves. Let's open an issue for migrating to Material 3!